### PR TITLE
Ests 131301 message aggregators

### PR DESCRIPTION
**Pull request by RobertDubiszDell**
- [ ] I have read the [CONTRIBUTION GUIDE][contributing]

- [ ] This is a bug

- [X] This is a feature request

- [ ] This is a syntax/style fix

- [X] This is an improvement to code quality

---
**Please describe your changes below:**
System Definition + Credentials + Identity message aggregators common code moved from rackhd-adapter to system-credentials library and the API independent part moved to common-messaging.

[contributing]: http://dellemc-symphony.readthedocs.io/en/latest/contributingtosymphony.html

### CODE REVIEW

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/AbstractAggregate.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+     */
+    protected boolean containsStatus(final CorrelationStatus status)
+    {
+        synchronized (this.correlationStates)
+        {
+            return this.correlationStates.containsValue(status);
+        }
+    }
+
+    @Override
```
Comment (ocallm1) [inline common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/AbstractAggregate.java]: add javadoc ?

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/AggregatorWrapper.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    void forceCompletion(final String initialRequestCorrelationId);
+
+    /**
+     * This returns true of the correlation identifier is being tracked by an aggregator.
+     *
+     * @param initialRequestCorrelationId The initial correlation identifier.
+     * @throws Exception Thrown if the message cannot be processed.
+     * @since 2.1.2
+     */
+    boolean checkIfCorrelationIdPresent(final String initialRequestCorrelationId)
```
Comment (ocallm1) [inline common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/AggregatorWrapper.java]: should we have a common exception?

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/CorrelationStatus.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * @since 2.1.2
+ */
+public enum CorrelationStatus
+{
+    /**
+     * The request has been sent.
+     *
+     * @since 2.1.2
+     */
+    REQUESTED,
```
Comment (ocallm1) [inline common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/CorrelationStatus.java]: you might want to add a String mapping on these

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/AggregatorWrapper.java (hunk 2 above)
Comment (RobertDubiszDell) [reply common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/aggregate/AggregatorWrapper.java]: Implementation of this interface in system-credentials throws it's own Exception. Here it is just a generic interface.

**Pull request merged**